board.cpp

```diff
@@ -0,0 +1,164 @@
+            if (y == 0){ //si es la primera fila, imprimimos los numeros de las columnas
+                for(int space = 0; space < (3 - std::to_string(x).length()); space++){
+                    std::cout << " ";
+                };
+                std::cout << std::to_string(x);
+            }
+            else{ //en caso contrario, imprimimos el estado de la posicion (" " = no existe, "O" = posicion en juego/estado desconocido, "M" = mina del jugador en la posicion)
+                isPositionEnabled = true;
+                for(std::vector<Position>::iterator it = board.disabledPositions.begin(); it != board.disabledPositions.end(); it++){
+                    if(it->xpos == x && it->ypos == y){
+                        isPositionEnabled = false;
+                    }
+                }
+                bool mineInPos=false;
+                if (perspective != -1){ //shows the mines belonging to player
+                    for(std::vector<Mine>::iterator it = board.placedMines.begin(); it != board.placedMines.end(); it++){
+                        if(it->xpos == x && it->ypos == y){
+                            if(it->owner == perspective){
+                                std::cout << "  M";
+                                mineInPos=true;
+                            }
+                        }
+                    }
+                }
+                if(mineInPos == false){
+                    if(isPositionEnabled==false){
+                        std::cout << "   ";
+                    }
+                    else if(isPositionEnabled == true){
+                        std::cout << "  O";
+                    }
+                }
+            }
+        }
+        std::cout << std::endl;
+    }
+}
+
+void disablePosition(Board &board, Position disabledPosition){
+    board.disabledPositions.push_back(disabledPosition);
+}
+
+bool removeMine(Board &board, Mine mine){
+    for(std::vector<Mine>::iterator it = board.placedMines.begin(); it != board.placedMines.end(); it++){
+        if (*it == mine){
+            board.placedMines.erase(it);
+            board.players[mine.owner].mineCount--;
+            std::cout << "Player " << mine.owner + 1 << " mines remaining: " << board.players[mine.owner].mineCount << std::endl;
+            disablePosition(board, mine);
+            return true;
+        }
+    }
+    return false; //just in case I need to check if a mine was properly erased or not
+}
+
+//function to be called at the end of each turn
+void disableTilesUsed(Board &board){ 
+    for(std::vector<Mine>::iterator it = board.placedMines.begin(); it != board.placedMines.end(); it++){
+        disablePosition(board, *it);
+    }
+}
+
+//when only one player has mines remaining, they win the game
+//if no players have mines, game is a draw
+//TODO: refactor to support more than 2 players
+int checkWinCon(Board board){

```

Comment:
#should rename function, as this does not check the win condition, but returns a player number instead